gravitee-apim-console-webui/src/management/api/endpoints-v4/backend-services/endpoints-group/api-endpoints-group.component.spec.ts

```diff
@@ -0,0 +1,374 @@
+          maxConcurrentConnections: 100,
+          useCompression: true,
+          version: 'HTTP_1_1',
+          pipelining: false,
+        },
+        ssl: {
+          keyStore: {
+            type: '',
+          },
+          hostnameVerifier: false,
+          trustStore: {
+            type: '',
+          },
+          trustAll: false,
+        },
+      },
+    },
+  ],
+};
+
+function getExpectedEndpoints(api: ApiV4, endpointGroupAttributes): EndpointGroupV4[] {
+  return [{ ...api.endpointGroups[GROUP_INDEX], ...endpointGroupAttributes }];
+}
+
+/**
+ * Test data for a mock API V4 responses
+ */
+export const TEST_MOCK_API_REQUESTS = {
+  expectApiGetRequest(api: ApiV4, fixture: ComponentFixture<any>, httpTestingController) {
+    httpTestingController.expectOne({ url: `${CONSTANTS_TESTING.env.v2BaseURL}/apis/${api.id}`, method: 'GET' }).flush(api);
+    fixture.detectChanges();
+  },
+  expectApiPutRequest(api: ApiV4, fixture: ComponentFixture<any>, httpTestingController: HttpTestingController): TestRequest {
+    return httpTestingController.expectOne({ url: `${CONSTANTS_TESTING.env.v2BaseURL}/apis/${api.id}`, method: 'PUT' });
+  },
+  expectConnectorRequest(
+    fixture: ComponentFixture<any>,
+    httpTestingController: HttpTestingController,
+    httpConnector: ConnectorListItem,
+    grpcConnector: ConnectorListItem,
+  ) {
+    httpTestingController
+      .expectOne({ url: `${CONSTANTS_TESTING.env.baseURL}/connectors?expand=schema`, method: 'GET' })
+      .flush([httpConnector, grpcConnector]);
+    fixture.detectChanges();
+  },
+};
+
+/**
+ * Test data for a mock API V4 responses
+ */
+export const TEST_MOCK_API_RESPONSES = {
+  ENDPOINT_GROUPS_DEFAULT_API_DATA,
+};
+
+/**
+ * Test data for an Angular component inputs
+ */
+export const TEST_INPUT_DATA = {
+  API_ID,
+  GROUP_INDEX,
+  DEFAULT_GROUP_NAME,
+  VALID_ENDPOINT_GROUP_NAME,
+  INVALID_ENDPOINT_GROUP_NAME,
+  DEFAULT_LOAD_BALANCER_TYPE,
+  ALTERNATE_LOAD_BALANCER_TYPE,
+  ALTERNATE_LOAD_BALANCER_TYPE_2,
+};
+
+/**
+ * Test data for an Angular component outputs (a mock function)
+ */
+export const TEST_OUTPUT_DATA = {
+  FAKE_UI_ROUTER: { go: jest.fn() },
+};
+
+/**
+ * Test utility functions
+ */
+export const TEST_UTILITY_FUNCTIONS = {
+  getExpectedEndpoints,
+};
+
+describe('ApiEndpointsGroupsComponent', () => {
+  let fixture: ComponentFixture<ApiEndpointsGroupComponent>;
+  let httpTestingController: HttpTestingController;
+  let componentHarness: ApiEndpointsGroupHarness;
+  let httpConnector: ConnectorListItem;
+  let grpcConnector: ConnectorListItem;
+
+  const initComponent = async () => {
+    const currentUser = new DeprecatedUser();
+
+    currentUser.userPermissions = ['api-definition-u', 'api-definition-c', 'api-definition-r'];
+
+    const routerParams: unknown = { apiId: TEST_INPUT_DATA.API_ID, groupIndex: TEST_INPUT_DATA.GROUP_INDEX };
+
+    const api = fakeApiV4(TEST_MOCK_API_RESPONSES.ENDPOINT_GROUPS_DEFAULT_API_DATA);
```

Comment:
You can simplify a bit by using it like that
```suggestion
    const api = fakeApiV4({id: API_ID, endpointGroups: TEST_MOCK_API_RESPONSES.ENDPOINT_GROUPS_DEFAULT_API_DATA});
```

This way, `ENDPOINT_GROUPS_DEFAULT_API_DATA` can contain only endpoint groups data and you can avoid declaring DEFAULT_MOCK_API_DATA_MODEL type just for that.